### ISSUE TITLE
Handling flagged data

### ISSUE THREAD

**Issue by ryanaloomis** [AstroChem/vis_sample#23]
Need to properly parse and handle flagged data.

At least 3 places this is needed:
1. In the initial import_data step - the user likely does not want interpolated data for flagged visibilities to be returned.
2. In diff mode - flagged visibilities should not be returned as part of the residuals
3. When writing back to a measurement set - visibilities need to be placed properly back into the ms if it originally contained flagged data. Flagged visibilities should ideally be set to 0 and a note placed in the header (see issue #16 )

**Comment by j6626** [comment]
one suggestion - 

As Ian pointed out in #24, one of the main difficulties with dealing with flagged data in vis_sample is that if you remove the data in the import_data step, you end up with an array that doesn't match up in shape with the data, which would make the comparison calculations annoying if not impossible. 

Perhaps one way to get around this problem is not to create new arrays, but simply to create a boolean mask and return that to the user, who can then apply it to the data and model to exclude the flagged data. That's admittedly a bit annoying, but presumably that is information that the user would have needed to do the chi^2 calculation anyway if they haven't already removed their flagged data.

**Comment by iancze** [comment]
That is essentially the compromise that I had to work with in UVHDF5 and
DiskJockey. Not the most seamless, but I can't think of anything simpler
than carrying the flags as a boolean mask.

**Comment by ral5bfloomis** [comment]
Yes, returning a boolean mask to the user definitely seems like a good
approach.

Given how users might implement the code, however, I'm not sure if this
would be better to return as a warning with the mask or if there might be a
better way to format the communication to the user. Ie an unexpected
returned mask might crash their wrapping code.

Thoughts? Is there a good way to default output this sort of information?

**Comment by j6626** [comment]
I suppose it's valid just to print a warning. There's nothing intrinsically wrong with interpolating onto flagged points. That being said, I also think one has to be willing to break backwards compatibility for a package that's still under development.

**Comment by ral5bfloomis** [comment]
A warning seems to be the most intuitive approach for the user. I'll put
that on the to do list unless anyone has objections...

**Comment by j6626** [comment]
I already wrote one. I'll just push it directly to master.

**Comment by ral5bfloomis** [comment]
Perfect!
